### PR TITLE
Add unit tests for error estimation

### PR DESCRIPTION
**Pull request by jwallwork23**
I'm making an effort to improve the test coverage across Pyroteus and hopefully improving its robustness in the process. Here are some tweaks to the error estimation code, plus unit tests.

Partially addresses #77.

### CODE REVIEW

File: pyroteus/error_estimation.py [hunk 1]
```diff
@@ -4,40 +4,39 @@
...
 
 
-@PETSc.Log.EventDecorator("pyroteus.form2indicator")
+@PETSc.Log.EventDecorator()
 def form2indicator(F) -> Function:
     """
-    Multiply throughout in a form and
-    assemble as a cellwise error
-    indicator.
+    Multiply throughout in a form and assemble as a cellwise error indicator.
```
Comment (jwallwork23) [inline pyroteus/error_estimation.py:20]: Be specific: 0-form

File: pyroteus/error_estimation.py [hunk 2]
```diff
@@ -50,113 +49,121 @@ def form2indicator(F) -> Function:
...
-    """
-    indicator = form2indicator(F)
-    return indicator2estimator(indicator, **kwargs)
-
-
-@PETSc.Log.EventDecorator("pyroteus.get_dwr_indicator")
-def get_dwr_indicator(F, adjoint_error: Function, **kwargs) -> Function:
+@PETSc.Log.EventDecorator()
+def get_dwr_indicator(
+    F, adjoint_error: Function, test_space: Optional[Union[WithGeometry, Dict]] = None
```
Comment (jwallwork23) [inline pyroteus/error_estimation.py:111]: WithGeometry?

File: pyroteus/error_estimation.py [hunk 3]
```diff
@@ -50,113 +49,121 @@ def form2indicator(F) -> Function:
...
+) -> Function:
     """
-    Generate a dual weighted residual (DWR)
-    error indicator, given a form and an
-    approximation of the error in the adjoint
-    solution.
+    Generate a dual weighted residual (DWR) error indicator, given a form and an
```
Comment (jwallwork23) [inline pyroteus/error_estimation.py:114]: Note that it maps 1-form to 0-form. Provide more information.

File: pyroteus/error_estimation.py (hunk 2 above)
Comment (jwallwork23) [reply pyroteus/error_estimation.py:111]: Okay so it seems `FunctionSpace` is actually a function that creates a `WithGeometry`.

**Review by stephankramer — APPROVED** [review]

**Pull request merged**